azbus/receiver.go

```diff
@@ -340,25 +341,32 @@ func (r *Receiver) open() error {
 			return fmt.Errorf("failed to open handler: %w", err)
 		}
 	}
+	ctx, cancel := context.WithCancel(context.Background())
+	r.ctx = ctx
+	r.cancel = cancel
 	return nil
 }
 
 func (r *Receiver) close_() {
 	if r != nil {
+		r.log.Debugf("Close")
 		if r.receiver != nil {
 			r.mtx.Lock()
 			defer r.mtx.Unlock()
 
+			for j := range len(r.handlers) {
+				r.log.Debugf("Close handler")
+				r.handlers[j].Close()
+			}
+
+			r.log.Debugf("Close receiver")
 			err := r.receiver.Close(context.Background())
 			if err != nil {
 				azerr := fmt.Errorf("%s: Error closing receiver: %w", r, NewAzbusError(err))
 				r.log.Infof("%s", azerr)
 			}
-			r.receiver = nil
-			for j := range len(r.handlers) {
-				r.handlers[j].Close()
-			}
 			r.handlers = []Handler{}
```

Comment:
Should we cancel and nil out the stashed Context here?

Reply:
yes fixed